spec/requests/api/v2/providers/publish_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'rails_helper'
+
+describe 'Provider Publish API v2', type: :request do
+  let(:user)         { create(:user) }
+  let(:organisation) { create(:organisation, users: [user]) }
+  let(:provider)     { create :provider, organisations: [organisation] }
+  let(:payload)      { { email: user.email } }
+  let(:token)        { build_jwt :apiv2, payload: payload }
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+
+  describe 'POST publish' do
+    let(:publish_path) do
+      "/api/v2/recruitment_cycles/#{provider.recruitment_cycle.year}" +
+        "/providers/#{provider.provider_code}/publish"
+    end
+
+    let(:enrichment) { build(:provider_enrichment, :initial_draft) }
+
+    subject do
+      post publish_path,
+           headers: { 'HTTP_AUTHORIZATION' => credentials },
+           params: {
+             _jsonapi: {
+               data: {
+                 attributes: {},
+                 type: "provider"
+               }
+             }
+           }
+      response
+    end
+
+    context 'when unauthenticated' do
```

Comment:
We should DRY up these specs if they're needed across multiple tests.

Reply:
Will do 👍